frontend/ui/widget/iconbutton.lua

```diff
@@ -31,25 +38,42 @@ function IconButton:init()
 
     self.show_parent = self.show_parent or self
 
-    self.button = HorizontalGroup:new{}
-    table.insert(self.button, HorizontalSpan:new{})
-    table.insert(self.button, self.image)
-    table.insert(self.button, HorizontalSpan:new{})
+    self.horizontal_group = HorizontalGroup:new{}
+    table.insert(self.horizontal_group, HorizontalSpan:new{})
+    table.insert(self.horizontal_group, self.image)
+    table.insert(self.horizontal_group, HorizontalSpan:new{})
+
+    self.button = VerticalGroup:new{}
+    table.insert(self.button, VerticalSpan:new{})
+    table.insert(self.button, self.horizontal_group)
+    table.insert(self.button, VerticalSpan:new{})
 
     self[1] = self.button
     self:update()
 end
 
 function IconButton:update()
-    self.button[1].width = self.horizontal_padding
-    self.button[3].width = self.horizontal_padding
+    if self.horizontal_padding > 0 then self:setHorizontalPadding(self.horizontal_padding) end
```

Comment:
Looks fine. May be you could just get rid of self.horizontal_padding, and use padding_left and padding_right in touchmenu.

Reply:
Yes, I just wanted this to be mergeable.

Reply:
@poire-z Is there a particular reason you didn't just do this?

```lua
            menu_items[c].padding_left = padding
            menu_items[c].padding_right = padding
            -- or rather horizontal_padding
            menu_items[c]:update()
```

Reply:
Is this still a question ? (hard to follow your thoughts with the multiple crosslinks :)
If yes, well, I didnt because they weren't there cause you just added those padding_left & co :)
And before that, I would have had to do instead:
```
menu_items[c].button[1].width = padding
menu_items[c].button[3].width = padding
menu_items[c]:update()
```
which expose too much of the internals.
Or I misunderstood your question ?

Reply:
I was lazy and copied my new code. I just meant

menu_items[c].horizontal_padding = padding
menu_items[c]:update()

Anyway, if you have no objections I reckon this is ready to be merged?

Reply:
OK. My thinking was just to not expose the existence of the internal update(), and provide some public method setHorizontalPadding() for that.
But in your case, as padding_left & co are known, I guess it's ok for update() to be known and called. So no objection to merge.